src/cfml/system/endpoints/Maven.cfc

```diff
@@ -0,0 +1,701 @@
+		orderedStruct[ "jitpack" ]         = "https://jitpack.io/";
+		orderedStruct[ "google" ]          = "https://maven.google.com/";
+		orderedStruct[ "spring" ]          = "https://repo.spring.io/release/";
+		orderedStruct[ "jboss" ]           = "https://repository.jboss.org/nexus/content/repositories/releases/";
+		orderedStruct[ "apacheSnapshots" ] = "https://repository.apache.org/snapshots/";
+		orderedStruct[ "gradlePlugins" ]   = "https://plugins.gradle.org/m2/";
+		setGlobalRepos( orderedStruct );
+		return this;
+	}
+
+	/**
+	 * Resolves the Maven package based on the provided package string.
+	 * Handles different URL patterns for Maven repositories.
+	 * @package The package to resolve
+	 * @currentWorkingDirectory The directory to resolve the package in
+	 * @verbose Verbose flag or silent, defaults to false
+	 */
+	public string function resolvePackage(
+		required string package,
+		string currentWorkingDirectory = "",
+		boolean verbose                = false
+	){
+		var job   = wirebox.getInstance( "interactiveJob" );
+		var repos = getGlobalRepos(); // Global linked struct of repos
+		// var projectRepos = {}; // TODO: Local overrides from box.json
+		// var repos = structAppend(globalRepos, projectRepos, false); // Preserve order
+
+		var artifactParts = getArtifactParts( package );
+		var jarFileURL    = "";
+		var artifact      = {
+			"jarFileURL"       : "",
+			"artifactMetadata" : {}
+		};
+
+		// If the local artifact exists, serve it
+		if (
+			artifactService.artifactExists( artifactParts.artifactId, artifactParts.version ) && artifactParts.version != "STABLE" && !semanticVersion.isExactVersion(
+				artifactParts.version,
+				true
+			)
+		) {
+			job.addLog( "Lucky you, we found this version in local artifacts!" );
+			var thisArtifactPath = artifactService.getArtifactPath( artifactParts.artifactId, artifactParts.version );
+
+			// Return the path to the artifact
+			return fileEndpoint.resolvePackage(
+				thisArtifactPath,
+				currentWorkingDirectory,
+				arguments.verbose
+			);
+		}
+
+		// Check only the explicitly defined repo, if any
+		if ( len( artifactParts.repo ) ) {
+			artifact = getArtifactFromRepo(
+				artifactParts.repo,
+				artifactParts.groupId,
+				artifactParts.artifactId,
+				artifactParts.version
+			);
+			jarFileURL = artifact.jarFileURL;
+		}
+		// Otherwise, check each registered repo sequentially
+		else {
+			for ( var alias in repos ) {
+				artifact = getArtifactFromRepo(
+					repos[ alias ],
+					artifactParts.groupId,
+					artifactParts.artifactId,
+					artifactParts.version
+				);
+				jarFileURL = artifact.jarFileURL;
+				// If we found the artifact, break out of the loop
+				if ( jarFileURL.len() ) {
+					break;
+				}
+			}
+		}
+
+		// Defer to jar endpoint
+		var folderName = jarEndpoint.resolvePackage(
+			artifact.jarFileURL,
+			currentWorkingDirectory,
+			arguments.verbose
+		);
+
+		if ( artifactParts.version eq "STABLE" ) {
```

Comment:
Why are we looking up the latest version after we've already downloaded the jar above?  This seems out of order.  All the version resolution should be done before this step.